iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/port_introspection.hpp

```diff
@@ -416,7 +415,7 @@ class PortIntrospection
  * @brief typedef for the templated port introspection class that is used by RouDi for the
  * actual port introspection functionality.
  */
-using PortIntrospectionType = PortIntrospection<SenderPortType, ReceiverPortType>;
+using PortIntrospectionType = PortIntrospection<PublisherPortUserType, SubscriberPortUserType>;
```

Comment:
+1 for making all the introspection classes template free, storing just a `PublisherPortData*` is better IMHO.